Records that contain Enum are still unsupported with @ParameterizedTest even with Serializable in 3.13.0

### Describe the bug

This is kind of this [#40148](https://github.com/quarkusio/quarkus/issues/40148) , [#42006](https://github.com/quarkusio/quarkus/issues/42006) and[#42098](https://github.com/quarkusio/quarkus/issues/42098).
When a @ParameterizedTest contains a record as a parameter that has an enum property the error is thrown like this
`[ERROR]   GreetingResourceTest.should_postUsers(UserRequest)[1] » IllegalState Error calling onstructor on record class class org.acme.model.UserRequest` if remove enum then it starts to work.

PS. But the same test works fine with Serializable as expected  in 3.12.3 

### Expected behavior

Each @ParameterizedTest must accept a `record` containing a property of `Enum<E extends Enum<E>>` type

### Actual behavior

Each @ParameterizedTest is failed if a parameter is a record that contains enum property

### How to Reproduce?

Create any `@ParameterizedTest` with `@MethodSource("requestSamples")` where the method `requestSamples ` provides a `Stream<SomeRecord>` and `SomeRecord ` looks like
```
public record SomeRecord(String name, SomeEnum someEnum) { 
    enum SomeEnum { A, B, C }
}
```
here is some project example 
[code-with-quarkus.zip](https://github.com/user-attachments/files/16473049/code-with-quarkus.zip)



### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -35,6 +35,7 @@ updates:
       - dependency-name: org.glassfish:jakarta-el
       - dependency-name: com.google.cloud.tools:jib-core
       - dependency-name: org.jboss.threads:jboss-threads
+      - dependency-name: org.jboss.marshalling:*
       # Quarkus
       - dependency-name: io.quarkus.*:*
       - dependency-name: io.quarkus:*

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -112,7 +112,7 @@
         <wildfly-common.version>1.7.0.Final</wildfly-common.version>
         <wildfly-client-config.version>1.0.1.Final</wildfly-client-config.version>
         <wildfly-elytron.version>2.5.2.Final</wildfly-elytron.version>
-        <jboss-marshalling.version>2.1.4.SP1</jboss-marshalling.version>
+        <jboss-marshalling.version>2.2.0.Final</jboss-marshalling.version>
         <jboss-threads.version>3.6.1.Final</jboss-threads.version>
         <vertx.version>4.5.9</vertx.version>
         <httpclient.version>4.5.14</httpclient.version>

